### PR TITLE
Shooter subsystem

### PR DESCRIPTION
**Pull request by wilsonwatson**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package frc.robot.subsystems.shooter;
+
+import com.revrobotics.CANSparkFlex;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import frc.robot.Constants;
+
+public class ShooterVortex implements ShooterIO {
```
Comment (wilsonwatson) [inline src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java:7]: This class is missing an implementation of `updateInputs`.

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package frc.robot;
 
+
+
```
Comment (legoguy1000) [inline src/main/java/frc/robot/Constants.java:4]: remove extra lines

File: src/main/java/frc/robot/Constants.java [hunk 3]
```diff
@@ -28,6 +30,13 @@ public static final class Motors {
      */
     public static final class Pneumatics {
     }
+    public static final class Shooter {
+        public static final int shooterTopId = 0;
+        public static final int shooterBottomId = 0;
```
Comment (legoguy1000) [inline src/main/java/frc/robot/Constants.java:35]: Move Motor CAN Ids into the Motors class above.  See whats currently in main

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    private ShooterIO io;
+    private PIDController pid =
+        new PIDController(Constants.Shooter.kP, Constants.Shooter.kI, Constants.Shooter.kD);
+    private ShooterIOInputsAutoLogged inputs = new ShooterIOInputsAutoLogged();
+
+    public Shooter(ShooterIO io) {
+        this.io = io;
+    }
+
+    public void periodic() {
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:18]: This should be an `@Override`

File: src/main/java/frc/robot/subsystems/shooter/ShooterIO.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    }
+
+    public default void setBottomMotor(double power) {
+
+    }
+
+    public default void updateInputs(ShooterIOInputsAutoLogged inputs) {
+
+    }
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/shooter/ShooterIO.java:24]: collapse to 1 line

File: src/main/java/frc/robot/subsystems/shooter/ShooterIO.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    public default void setTopMotor(double power) {
+
+    }
+
+    public default void setBottomMotor(double power) {
+
+    }
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/shooter/ShooterIO.java:20]: collapse to 1 line

File: src/main/java/frc/robot/subsystems/shooter/ShooterIO.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    @AutoLog
+    public static class ShooterIOInputs {
+        public double shooterVelocityRotPerSecond;
+
+
+    }
+
+    public default void setTopMotor(double power) {
+
+    }
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/shooter/ShooterIO.java:16]: collapse to 1 line

File: src/main/java/frc/robot/subsystems/shooter/ShooterIO.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import org.littletonrobotics.junction.AutoLog;
+
+public interface ShooterIO {
+
+    @AutoLog
+    public static class ShooterIOInputs {
+        public double shooterVelocityRotPerSecond;
+
+
+    }
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/shooter/ShooterIO.java:12]: remove extra blank lines

File: src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        shooterTopMotor.setVoltage(power);
+    }
+
+    public void setBottomMotor(double power) {
+        shooterBottomMotor.setVoltage(power);
+    }
+
+    public void updateInputs(ShooterIOInputs inputs) {
+        inputs.topshooterVelocityRotPerSecond = topEncoder.getPosition();
+        inputs.bottomshooterVelocityRotPerSecond = bottomEncoder.getPosition();
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/shooter/ShooterVortex.java:31]: `.getPosition()` gives you the position not velocity.

**Review by legoguy1000 — CHANGES_REQUESTED** [review]

**Review by legoguy1000 — APPROVED** [review]

**Review by legoguy1000 — CHANGES_REQUESTED** [review]
Add commands

**Review by legoguy1000 — CHANGES_REQUESTED** [review]
You need to add subsystem to Robot Container

File: src/main/java/frc/robot/RobotContainer.java [hunk 10]
```diff
@@ -29,6 +30,7 @@ public class RobotContainer {
 
     /* Subsystems */
     private Drivetrain drivetrain;
+    private Shooter shooter;
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java:33]: Look at Robot Container COntrstuctor!!!!!!

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+     */
+    public Command runShooterMotor() {
+        return Commands.startEnd(() -> {
+            setTopMotor(0.5);
+            setBottomMotor(0.5);
+        }, () -> {
+            setTopMotor(0);
+            setBottomMotor(0);
+        }, this);
+    }
```
Comment (legoguy1000) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:56]: This doesn't make sense.. You're not using the PID controller at all here. Also the PID doesn't need to be running all the time. Only when we want to shoot, ie when I start the command.

**Review by legoguy1000 — APPROVED** [review]

**Pull request merged**